axelrod/tests/strategies/test_defector.py

```diff
@@ -61,3 +61,21 @@ def test_defects_if_opponent_last_three_are_not_D(self):
         self.versus_test(
             axl.MockPlayer(actions=opponent_actions), expected_actions=actions
         )
+
+
+class TestModalDefector(TestPlayer):
+    name = "Modal Defector"
+    player = axl.ModalDefector
+    expected_classifier = {
+        "memory_depth": float("inf"),
+        "stochastic": False,
+        "makes_use_of": set(),
+        "inspects_source": False,
+        "manipulates_source": False,
+        "manipulates_state": False,
+    }
+
+    def test_strategy(self):
```

Comment:
Try to add a few more tests that single out the specific behaviors like the first move and that the mode works for both C and D. If possible try to use a real strategy like Alternator rather than mock players, but mocks are ok if it's hard to target a particular case.